src/lib.rs

```diff
@@ -258,12 +261,27 @@ impl OfferHandler {
             reply_path,
         };
 
-        let mut pending_messages = self.pending_messages.lock().unwrap();
-        pending_messages.push(pending_message);
-        std::mem::drop(pending_messages);
+        {
+            let mut pending_messages = self.pending_messages.lock().unwrap();
+            pending_messages.push(pending_message);
 
-        let mut active_offers = self.active_offers.lock().unwrap();
-        active_offers.insert(offer.to_string().clone(), OfferState::InvoiceRequestSent);
+            let mut active_offers = self.active_offers.lock().unwrap();
+            active_offers.insert(offer.to_string().clone(), OfferState::InvoiceRequestSent);
+        }
+
+        let invoice = self.wait_for_invoice().await;
+        let payment_hash = invoice.payment_hash();
+        let path_info = invoice.payment_paths()[0].clone();
+
+        let _ = pay_invoice(
+            client,
+            path_info.1,
+            path_info.0.cltv_expiry_delta,
+            path_info.0.fee_base_msat,
+            payment_hash.0,
+            amount.unwrap(),
+        )
+        .await;
```

Comment:
As in comment on previous commit, the payment hasn't actually finished when this returns.